src/Trie.js

```diff
@@ -153,19 +120,19 @@ class Trie {
       node = node.socket[char]
     }
 
-    node.name = key
+    node.key = key
     node.value = value
 
-    this.cache.set(key, node)
+    super.set(key, node)
   }
 
   /**
-    Removes any value associated to the key and returns the value that has(key) would have previously returned. has(key) will return false afterwards
+    Delete any value associated to the key and returns the value that has(key) would have previously returned. has(key) will return false afterwards
 
-    @example store.remove('Michael Jacobs')
+    @example store.delete('Michael Jacobs')
   */
   delete (query: string): void {
```

Comment:
From MDN about `delete`:

> Returns true if an element in the Map object existed and has been removed, or false if the element does not exist.

Check that your methods does exactly that and test it